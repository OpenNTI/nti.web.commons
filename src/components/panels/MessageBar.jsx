import './MessageBar.scss';
import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

MessageBar.propTypes = {
	message: PropTypes.string,
	error: PropTypes.bool,
	iconCls: PropTypes.string
};

export default function MessageBar ({message, error, iconCls}) {
	const cls = cx('panels-message-bar', {error});

	return (
		<div className={cls}>
			{iconCls && (<i className={iconCls} />)}
			{message}
		</div>
	);
}
