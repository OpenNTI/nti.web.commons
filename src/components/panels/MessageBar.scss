@import "~@nti/style-common";

.panels-message-bar {
	background: $panel-background-yellow;
	padding: rem-calc(8px 30px);
	font: normal 400 0.875rem/1.25rem $body-font-family;
	color: var(--primary-grey);
	box-shadow: inset 0 -1px 0 0 $border-grey-light, inset 0 1px 0 0 $border-grey-light;

	i {
		margin-right: 0.5rem;
		vertical-align: middle;
	}

	&.error {
		background: var(--primary-red);
		color: white;
	}
}
