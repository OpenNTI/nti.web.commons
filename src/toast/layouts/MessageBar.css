.toast-message-bar {
	border-radius: 3px;
	background-color: var(--panel-background);
	box-shadow: 0 0 0 1px var(--border-grey-light);
	padding: 0.25rem 0.5rem;
	display: flex;
	flex-direction: row;
	align-items: flex-start;
}

.toast-message-bar.error {
	background-color: var(--primary-red);
	box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.3);

	.icon-container {
		color: white;
	}

	.message-container {
		color: white;
	}

	.close-icon {
		color: white;
	}
}

.icon-container {
	flex: 0 0 auto;
	align-self: center;
	margin-right: 0.25rem;
	color: var(--secondary-grey);
}

.message-container {
	flex: 1 1 auto;
	font-size: 0.875rem;
	color: var(--secondary-grey);
}

.title {
	font-weight: 600;
	margin-right: 0.5rem;
}

.message {
	font-weight: 400;
}

.close-icon {
	flex: 0 0 auto;
	color: var(--secondary-grey);
}
