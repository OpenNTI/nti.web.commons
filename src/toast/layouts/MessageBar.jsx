import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';

import Text from '../../text';

import Styles from './MessageBar.css';

const cx = classnames.bind(Styles);

ToastMessage.propTypes = {
	className: PropTypes.string,
	error: PropTypes.bool,

	icon: PropTypes.node,

	title: PropTypes.string,
	message: PropTypes.string,

	onDismiss: PropTypes.func,
};
export default function ToastMessage({ className, error, icon, title, message, onDismiss }) {
	return (
		<div className={cx('toast-message-bar', className, {error})} onClick={onDismiss}>
			{icon && (<div className={cx('icon-container')}>{icon}</div>)}
			<div className={cx('message-container')}>
				{title && (
					<Text.Base className={cx('title')}>{title}</Text.Base>
				)}
				{message && (
					<Text.Base className={cx('message')}>{message}</Text.Base>
				)}
			</div>
			{onDismiss && <i className={cx('icon-bold-x', 'close-icon')} />}
		</div>
	);
}
