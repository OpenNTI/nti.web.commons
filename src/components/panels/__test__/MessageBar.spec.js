import React from 'react';
import renderer from 'react-test-renderer';

import MessageBar from '../MessageBar';

/* eslint-env jest */
describe('Panels: Message Bar', () => {
	test('Generic Snapshot', () => {
		const tree = renderer.create(<MessageBar message="hey it's me ya boi" />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('Error Snapshot', () => {
		const tree = renderer.create(<MessageBar message="Unable to upload file." error />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
