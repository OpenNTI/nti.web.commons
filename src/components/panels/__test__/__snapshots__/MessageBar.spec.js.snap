// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panels: Message Bar Error Snapshot 1`] = `
<div
  className="panels-message-bar error"
>
  Unable to upload file.
</div>
`;

exports[`Panels: Message Bar Generic Snapshot 1`] = `
<div
  className="panels-message-bar"
>
  hey it's me ya boi
</div>
`;
